### PR TITLE
Chore/ch71957/migrate client to poetry

### PR DESCRIPTION
**Pull request by quentinv72**
## Purpose

Migrate from setuptools to Poetry. This will allow us to transfer ownership of this repository to Cornershop and then deploy the package in our internal python index.

I also split up some of the tests into different modules because the code was split between different modules inside of the package. 

Finally, I modified the CI so that it can run the tests in Poetry. This is a temporary modification until we can ask infrastructure team to set up a proper CI/CD pipeline.

## Approach

Followed the first 3 steps in the tutorial I wrote on Notion.

## This PR addresses a:

- [ ] Bug
- [ ] Feature
- [x] Chore

## Quality assurance
The tests pass.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,3 +1,12 @@
 import os
 
+import pytest
+
+from cornershop_catalog_integrations_api_client import client, endpoint
+
 os.environ["BASE_API_URL"] = "https://catalog.superpal.com/api/v101"
```
Comment (emjironal) [inline tests/conftest.py:7]: This is not longer necessary since we now use two different Endpoints one for QA and Prod

Comment (quentinv72) [reply tests/conftest.py:7]: I will fix that thanks

**Comment by uchenna-cshop** [comment]
LGTM

**Review by uchenna-cshop — APPROVED** [review]
LGTM

**Review by emjironal — APPROVED** [review]

**Pull request merged**